do_files_push.go

```diff
@@ -284,19 +279,19 @@ func doFilesPush(
 }
 
 func returnError(err error) bool {
-	if err.Error() == "failed to upload original file: unable to authenticate: "+
-		"authentication parameters are invalid" {
+	if errors.Is(err, smartling.NotAuthorizedError{}) {
 		return true
 	}
 
-	reasons := []string{
-		"AUTHENTICATION_ERROR",
-		"AUTHORIZATION_ERROR",
-		"MAINTENANCE_MODE_ERROR",
+	if (errors.Is(err, smartling.APIError{})) {
+		reasons := map[string]struct{}{
+			"AUTHENTICATION_ERROR":   {},
+			"AUTHORIZATION_ERROR":    {},
+			"MAINTENANCE_MODE_ERROR": {},
+		}
+		_, ok := reasons[err.(smartling.APIError).Code]
```

Comment:
Sorry, but i didn't get what we do in this line and what actually will be in `ok`

Reply:
This line does the following:

Convert err to type smartling.APIError (will make that clearer in a pr update)
ok is the result of check that a key equal to the err.Code field is present in reasons (php equivalent would be $ok = in_array($err["Code"], ["AUTHENTICATION_ERROR", "AUTHORIZATION_ERROR", MAINTENANCE_MODE_ERROR"], but go doesn't seem to have a built like in_array)